### PR TITLE
feat: Use Vitest instead of Jest

### LINKED ISSUE #263: Use Vitest intead of Jest

### PR DESCRIPTION
**Pull request by siketyan**
Closes #263 

This pull request replaces Jest with Vitest.

At the first try, I found that Vitest does not resolve `await-lock` module correctly.
The package is not maintained recently and has an issue with ESM interop.
I replaced it with [`async-mutex`](https://www.npmjs.com/package/async-mutex) to resolve the issue.

### CODE REVIEW

**Comment by mizdra** [comment]
Thank you for doing this work! I will accept and use this Pull Request.

> At the first try, I found that Vitest does not resolve await-lock module correctly.
> The package is not maintained recently and has an issue with ESM interop.
> I replaced it with [async-mutex](https://www.npmjs.com/package/async-mutex) to resolve the issue.

There is nothing against replacing it with `async-mutex`. However, I don't think that the package is not maintained. The maintainer replied to the issue comment in August 2023. It seems that the package is maintained.

File: jest.config.mjs [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-  extensionsToTreatAsEsm: ['.ts'],
-  moduleNameMapper: {
-    // Map `./**/xxx.js` to `./**/xxx` (for ESM)
-    '^(\\.{1,2}/.*)\\.js$': '$1',
-  },
-  transform: {
-    '^.+\\.(t|j)sx?$': '@swc/jest',
-  },
-  // for ESM
-  resolver: join(dir, 'packages/happy-css-modules/src/test-util/jest/resolver.cjs'),
```
Comment (mizdra) [inline jest.config.mjs:22]: SHOULD: Please remove `packages/happy-css-modules/src/test-util/jest/resolver.cjs`.

File: packages/happy-css-modules/src/test-util/util.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { sleepSync } from '../util.js';
 export const FIXTURE_DIR_PATH = resolve(
   realpathSync(tmpdir()),
   'happy-css-modules/fixtures',
-  process.env['JEST_WORKER_ID']!,
+  process.env['VITEST_WORKER_ID']!,
```
Comment (mizdra) [inline packages/happy-css-modules/src/test-util/util.ts:13]: MAY: `VITEST_WORKER_ID` is also fine, but `VITEST_POOL_ID` is better. `VITEST_POOL_ID` is an environment variable that has the same meaning as `JEST_WORKER_ID`.

https://vitest.dev/guide/migration#envs

**Comment by mizdra** [comment]
Should we also delete the following line?

https://github.com/mizdra/happy-css-modules/blob/b0cafcddfd0bf113d1f2b5f301e18742abd99873/eslint.config.js#L90

File: jest.config.mjs (hunk 1 above)
Comment (siketyan) [reply jest.config.mjs:22]: Removed in [7769515](https://github.com/mizdra/happy-css-modules/pull/288/commits/77695155a352484a0cee78b88a602d11e94fc5a9).

File: packages/happy-css-modules/src/test-util/util.ts (hunk 2 above)
Comment (siketyan) [reply packages/happy-css-modules/src/test-util/util.ts]: Replaced with VITEST_POOL_ID in [846bbb8](https://github.com/mizdra/happy-css-modules/pull/288/commits/846bbb8cf2041828305654cc5ea9a1d088cb6aa6).

**Comment by siketyan** [comment]
@mizdra 
Sorry for my miswording. I meant that `await-lock` does not ship with native ESM support, breaking the type definition of the default import.

**Comment by mizdra** [comment]
Let's ignore the Vite bug here. If you are interested in reporting this bug to Vite team, please do so. If you are not interested, I will do it for you.

**Review by mizdra — APPROVED** [review]
LGTM! Thank you for contributing!

**Comment by siketyan** [comment]
@mizdra I checked the spec and found that whitespaces before the sourceMappingURL comment are allowed, as defined in https://tc39.es/source-map/#extraction-javascript. Thus, I think it's an expected behaviour and we cannot declare this as a Vitest's bug.

**Comment by siketyan** [comment]
And, this example is similar to our situation: https://tc39.es/source-map/#example-5b7659a8. This type of code can be parsed as unintentional even if the regexp starts with ^, unless Vitest parses it using AST.

**Pull request merged**